### ISSUE TITLE
Android SDK publish stops after few seconds (30-40 seconds) enterperise edition

### ISSUE THREAD

**Issue by nagendraK6** [ant-media/Ant-Media-Server#2362]
### Short description
__Brief description of what happened__
I tried webrtc android sample app with enterprise edition of the server. After couple of seconds I see that publishing is getting stopped. On further investigation found that ice candidate receives the disconnect. On close analysis I found that stun server returned stun server binding request timed out and then entire ice connection disconnects. I tried with new stun server as wll (after hosting on aws) but faced the same problem

On the same android device if I try with webview js (in browser)  then no issue and there is no disconnect from server.

### Environment

* Operating system and version: Android 
* Java version: 
* Ant Media Server version: enterprise-2.0.0-20200504_1748
* Browser name and version: Android 8

### Steps to reproduce
1. install enterprise server
2. change server url and stream key in sample mode, change the mode to publishing mode
3. run on physical android device. After enabling permissions publishing starts but then 30-40 seconda later it stops.

### Expected behavior
__Put as much detail here as possible__
Ice connection shouldn't disconnect.

### Actual behavior
__Put as much detail here as possible__
Publishing should continue as long as user has not stopped it
 

### Logs
__Place logs on [pastebin](http://pastebin.com/) or elsewhere and put links here__

2020-07-21 18:25:31.530 12013-12055/io.antmedia.webrtc_android_sample_app E/stun_port.cc: (line 96): Binding request timed out from [0:0:0:x:x:x:x:x]:56421 (wlan0)
2020-07-21 18:25:31.638 12013-12055/io.antmedia.webrtc_android_sample_app E/stun_port.cc: (line 96): Binding request timed out from [0:0:0:x:x:x:x:x]:58209 (radio0)
2020-07-21 18:25:31.819 12013-12055/io.antmedia.webrtc_android_sample_app E/stun_port.cc: (line 96): Binding request timed out from [0:0:0:x:x:x:x:x]:39676 (lo)
2020-07-21 18:25:31.819 12013-12055/io.antmedia.webrtc_android_sample_app E/stun_port.cc: (line 96): Binding request timed out from 127.0.0.x:52523 (lo)

Join [Google Group](https://groups.google.com/forum/#!forum/ant-media-server) and be a part of Ant Media Server Community.

**Comment by nagendraK6** [comment]
It seems that websocket disconnected and there is no reconnection configuration for the websocket

**Comment by nagendraK6** [comment]
I debugged it further and tried with different websocket client (okhttp). Finally found that ICEConnection state is changing to disconnected and sometimes closed. On the server side, I found the error websocket enterprise error. No further details in logs.

Maybe somehow the server is getting a bad packet and then peer connection from the server is getting disconnected and that is what changes to ICEConnection disconnected

This issue is reported earlier as well. Any idea what could go wrong here? I and my cofounder are evaluating red5pro vs antmedia enterprise and the android client publishing is critical for us. Wondering if anyone can unblock on this.It will help us to evaluate the experience end to end.

cc: @mekya

**Comment by mekya** [comment]
Hey @nagendraK6 ,
Are you using our sample application for that? 

In normal cases, we don't encounter this case.  
I think I can make a test on my end with my android device and one of our test server in the cloud and then I'll update you.  

Thank you for reporting.

**Comment by nagendraK6** [comment]
Yes. It is with the sample application provided.

**Comment by mekya** [comment]
Hey @nagendraK6 ,
I made a quick test on my end. I've used real android device(Samsung S5, Android 6).
I've used a remote server(not a local one). After 5 minutes of non-stop publishing, I stop the streaming. 

I've just added a screenshot below. 
<img width="1353" alt="Screen Shot 2020-07-23 at 20 14 52" src="https://user-images.githubusercontent.com/3456251/88317205-9b386280-cd21-11ea-9a93-4724faf14250.png">

Basically problem may be something below. 
- ICE connection can be dropped if network has some kind of problems for packet delivery.  For this case, you need to try again. 

- If websocket is disconnected for any reason such as firewall, it also triggers the ice connection to disconnect.  For this case, just check that if your firewall disconnects the websocket connections after a specific timeout value.

**Comment by nagendraK6** [comment]
I have a few more observations. My video server is behind AWS load balancer. By default, the AWS application load balancer creates two instances of LB.  So let's say  URL is  **video.abc.com** then ping can return different ip addresses in round robin (IP of LB). I believe in this case when call starts maybe after some time different IP is detected as a result ICE connection state is updated and the callback is received.

**I tried without load balancer by directly mapping URL to video server IP and things work as expected.**

**Comment by nagendraK6** [comment]
Issue resolve. Root cause id is default AWS idle timeout. Once I modified it everything worked

**Comment by mekya** [comment]
Hi @nagendraK6 , 

Thank you for the feedback. There is also ping/pong messages in the SDK that handles this load balancer timeout. 

Could you please search in the SDK that has a method with `startPingPongTimer`? It should be in `WebSocketHandler.java`

**Comment by nagendraK6** [comment]
@mekya You are right. Ping/Pong is supposed to handle this. There is a pingPongTimoutCount that tracks any timeout and it is reset to 0 when we receive pong command. Looking at logs I can see that when aws LB used then the server is not returning pong message

2020-07-24 01:47:38.775 12985-12985/io.antmedia.webrtc_android_sample_app D/WSChannelRTCClient: Ping Pong timer is started
2020-07-24 01:47:41.776 12985-13150/io.antmedia.webrtc_android_sample_app D/WSChannelRTCClient: Ping Pong timer is executed
2020-07-24 01:47:41.777 12985-13150/io.antmedia.webrtc_android_sample_app W/WSChannelRTCClient: Ping/Pong message sent
2020-07-24 01:47:43.777 12985-13150/io.antmedia.webrtc_android_sample_app D/WSChannelRTCClient: Ping Pong timer is executed
2020-07-24 01:47:43.778 12985-13150/io.antmedia.webrtc_android_sample_app W/WSChannelRTCClient: Ping/Pong message sent
2020-07-24 01:47:45.776 12985-13150/io.antmedia.webrtc_android_sample_app D/WSChannelRTCClient: Ping Pong timer is executed
2020-07-24 01:47:45.777 12985-13150/io.antmedia.webrtc_android_sample_app W/WSChannelRTCClient: Ping/Pong message sent
2020-07-24 01:47:45.777 12985-13150/io.antmedia.webrtc_android_sample_app D/WSChannelRTCClient: Ping Pong websocket response not received for 4 seconds
2020-07-24 01:47:45.777 12985-13150/io.antmedia.webrtc_android_sample_app D/WSChannelRTCClient: Ping Pong timer stop called


I will check tomorrow if any port issue on my LB, which may be causing any issue. All other messages to exchange ICE is also going through the same websocket so should be fine.

**Comment by nagendraK6** [comment]
@mekya  **I think there is a small bug that may be on the server side**. The server is not sending the stream id in pong messages. As a result, the code hits exception and logic to validate that pong message is not executed and client stops the timer. Once the timer stops there is no ping message from the client and then a 60 seconds timer of AWS LB expires and the socket is disconnected.

**Comment by mekya** [comment]
You're great. Thank you very much. I found the problem in the SDK. I'm sending a fix for the SDK right now.
It'll be available to download later today

**Comment by mekya** [comment]
I've sent the fix and it'll available to download later today.
